src/pages/api/auth/[...auth0].ts

```diff
@@ -20,6 +45,39 @@ const handler: NextApiHandler = handleAuth({
       ...rest,
     });
   },
+  callback: async (req, res) => {
+    return handleCallback(req, res, {
+      afterCallback: async (callbackReq, callbackRes, session) => {
+        const { sub: id, email, name, picture, updatedAt } = session.user;
+
+        assertString(id, "id");
+        assertString(email, "email");
+        assertString(name, "name");
+        assertOptionalString(picture, "picture");
+        assertOptionalString(updatedAt, "updatedAt");
+
+        await prisma.user.upsert({
+          where: {
+            id,
+          },
+          update: {
+            name,
+            picture,
+            updatedAt,
+          },
+          create: {
+            email,
+            id,
+            name,
+            picture,
+            updatedAt,
+          },
+        });
```

Comment:
Once we’ve logged in, we upsert (update or insert) a user in our database. This eliminates the need to maintain all user endpoints we’ve previously had.